File: schemaorg-rs/src/bus_trip.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type BusTripAdditionalTypeFieldEnum = String;
///https://schema.org/departureTime
///https://schema.org/DateTime
///https://schema.org/Time
pub type BusTripDepartureTimeFieldEnum = String;
///https://schema.org/arrivalTime
///https://schema.org/DateTime
///https://schema.org/Time
pub type BusTripArrivalTimeFieldEnum = String;
///https://schema.org/BusTrip
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct BusTrip {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/tripOrigin
    #[serde(rename = "tripOrigin")]
    #[serde_as(as = "OneOrMany<_>")]
    pub trip_origin: Vec<Place>,
    ///https://schema.org/departureBusStop
    #[serde(rename = "departureBusStop")]
    #[serde_as(as = "OneOrMany<_>")]
    pub departure_bus_stop: Vec<BusTripDepartureBusStopFieldEnum>,
    ///https://schema.org/provider
    #[serde(rename = "provider")]
    #[serde_as(as = "OneOrMany<_>")]
    pub provider: Vec<BusTripProviderFieldEnum>,
    ///https://schema.org/partOfTrip
    #[serde(rename = "partOfTrip")]
    #[serde_as(as = "OneOrMany<_>")]
    pub part_of_trip: Vec<Trip>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/offers
    #[serde(rename = "offers")]
    #[serde_as(as = "OneOrMany<_>")]
    pub offers: Vec<BusTripOffersFieldEnum>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/busNumber
    #[serde(rename = "busNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub bus_number: Vec<String>,
    ///https://schema.org/arrivalBusStop
    #[serde(rename = "arrivalBusStop")]
    #[serde_as(as = "OneOrMany<_>")]
    pub arrival_bus_stop: Vec<BusTripArrivalBusStopFieldEnum>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<BusTripAdditionalTypeFieldEnum>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<BusTripImageFieldEnum>,
    ///https://schema.org/itinerary
    #[serde(rename = "itinerary")]
    #[serde_as(as = "OneOrMany<_>")]
    pub itinerary: Vec<BusTripItineraryFieldEnum>,
    ///https://schema.org/subTrip
    #[serde(rename = "subTrip")]
    #[serde_as(as = "OneOrMany<_>")]
    pub sub_trip: Vec<Trip>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<BusTripMainEntityOfPageFieldEnum>,
    ///https://schema.org/busName
    #[serde(rename = "busName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub bus_name: Vec<String>,
    ///https://schema.org/departureTime
    #[serde(rename = "departureTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub departure_time: Vec<BusTripDepartureTimeFieldEnum>,
    ///https://schema.org/arrivalTime
    #[serde(rename = "arrivalTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub arrival_time: Vec<BusTripArrivalTimeFieldEnum>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<BusTripIdentifierFieldEnum>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<BusTripDescriptionFieldEnum>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<BusTripSubjectOfFieldEnum>,
}

File: schemaorg-ts/src/bus_trip.ts
import type * as all from "./index";

/**
 * <https://schema.org/departureTime>
 **/
export type BusTripDepartureTimeFieldEnum = string;

/**
 * <https://schema.org/arrivalTime>
 **/
export type BusTripArrivalTimeFieldEnum = string;

/**
 * <https://schema.org/additionalType>
 **/
export type BusTripAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/BusTrip>
 **/
export interface BusTrip {
  "@context": string;
  "@type": "BusTrip";

  /**
   * <https://schema.org/departureBusStop>
   **/
  departureBusStop: all.OneOrMany<all.BusTripDepartureBusStopFieldEnum>;

  /**
   * <https://schema.org/arrivalBusStop>
   **/
  arrivalBusStop: all.OneOrMany<all.BusTripArrivalBusStopFieldEnum>;

  /**
   * <https://schema.org/busNumber>
   **/
  busNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/busName>
   **/
  busName: all.OneOrMany<string>;

  /**
   * <https://schema.org/provider>
   **/
  provider: all.OneOrMany<all.BusTripProviderFieldEnum>;

  /**
   * <https://schema.org/tripOrigin>
   **/
  tripOrigin: all.OneOrMany<all.Place>;

  /**
   * <https://schema.org/departureTime>
   **/
  departureTime: all.OneOrMany<all.BusTripDepartureTimeFieldEnum>;

  /**
   * <https://schema.org/partOfTrip>
   **/
  partOfTrip: all.OneOrMany<all.Trip>;

  /**
   * <https://schema.org/offers>
   **/
  offers: all.OneOrMany<all.BusTripOffersFieldEnum>;

  /**
   * <https://schema.org/itinerary>
   **/
  itinerary: all.OneOrMany<all.BusTripItineraryFieldEnum>;

  /**
   * <https://schema.org/arrivalTime>
   **/
  arrivalTime: all.OneOrMany<all.BusTripArrivalTimeFieldEnum>;

  /**
   * <https://schema.org/subTrip>
   **/
  subTrip: all.OneOrMany<all.Trip>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.BusTripMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.BusTripIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.BusTripDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.BusTripSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.BusTripAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.BusTripImageFieldEnum>;
}
